integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/vector/LuceneVectorFieldIT.java

```diff
@@ -0,0 +1,464 @@
+		assertThat( result )
+				.hasSize( 2 )
+				.containsOnly(
+						new Object[] { "keyword1", BYTE_VECTOR_1, FLOAT_VECTOR_1 },
+						new Object[] { "keyword2", BYTE_VECTOR_2, FLOAT_VECTOR_2 }
+				);
+	}
+
+	@Test
+	void simpleVectorPredicateNoFilter() {
+		SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimpleNoFilter( index );
+
+		assertThatQuery(
+				index.createScope().query()
+						.select( f -> f.field( "byteVector" ) )
+						.where( f -> f.knn( 3 ).field( "byteVector" )
+								.matching( bytes( 4, (byte) 5 ) ) )
+		).hasTotalHitCount( 3 )
+				.hasHitsAnyOrder(
+						bytes( 4, (byte) 4 ),
+						bytes( 4, (byte) 5 ),
+						bytes( 4, (byte) 6 )
+				);
+
+		assertThatQuery(
+				index.createScope().query()
+						.select( f -> f.field( "floatVector" ) )
+						.where( f -> f.knn( 3 ).field( "floatVector" )
+								.matching( floats( 8, 0.051f ) ) )
+		).hasTotalHitCount( 3 )
+				.hasHitsExactOrder(
+						floats( 8, 0.05f ),
+						floats( 8, 0.06f ),
+						floats( 8, 0.04f )
+				);
+	}
+
+	@ParameterizedTest
+	@EnumSource(VectorSimilarity.class)
+	void similarity(VectorSimilarity similarity) {
+		SimpleMappedIndex<SimilarityIndexBinding> index =
+				SimpleMappedIndex.of( root -> new SimilarityIndexBinding( similarity, root ) ).name( "index_" + similarity );
+		setupHelper.start().withIndexes( index ).setup();
+		initSimilarityIndexBinding( index );
+
+		assertThatQuery(
+				index.createScope().query()
+						.select( f -> f.id() )
+						.where( f -> f.knn( 1 ).field( "byteVector" )
+								.matching( bytes( 4, (byte) 5 ) ) )
+		).hasTotalHitCount( 1 );
+
+		assertThatQuery(
+				index.createScope().query()
+						.select( f -> f.id() )
+						.where( f -> f.knn( 1 ).field( "floatVector" )
+								.matching( floats( 8, 0.051f ) ) )
+		).hasTotalHitCount( 1 );
+	}
+
+	@Test
+	void simpleVectorPredicateWithFilter() {
+		// took the sample data and query from this example https://opensearch.org/docs/latest/search-plugins/knn/filter-search-knn/#using-a-lucene-k-nn-filter
+		// to see if we'll get the same results... and looks like we do :smile:
+		SimpleMappedIndex<PredicateIndexBinding> index = SimpleMappedIndex.of( PredicateIndexBinding::new )
+				.name( "predicateIndex" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimplePredicate( index );
+
+
+		int k = 3;
+		SearchQuery<float[]> query = index.createScope().query()
+				.select(
+						f -> f.field( "location", float[].class )
+				)
+				.where( f -> f.knn( k )
+						.field( "location" )
+						.matching( 5f, 4f )
+						.filter( f.range().field( "rating" ).between( 8, 10 ) )
+						.filter( f.terms().field( "parking" ).matchingAny( true ) )
+				).toQuery();
+
+		List<float[]> result = query.fetchAll().hits();
+
+		assertThat( result )
+				.hasSize( k ) // since that is how many neighbors we were asking for in the predicate
```

Comment:
Ah, that's interesting. What if you do `f.or( f.knn(3)..., f.somethingThatMatchesMoreThan3Documents() )`? Will you get more than 3 hits, as expected?

Reply:
```java
f.or(
		f.knn( 3 )
				.field( "location" )
				.matching( 5f, 4f ),
		f.terms().field( "parking" ).matchingAny( true )
)
```
yields:
```
[[5.2f, 4.4f],
    [4.9f, 3.4f],
    [7.0f, 9.9f],
    [6.4f, 3.4f],
    [2.4f, 4.0f],
    [3.3f, 4.5f],
    [5.0f, 1.0f],
    [4.2f, 6.2f],
    [5.2f, 3.9f]]
```
soo it kinda works 😃